javalib/src/main/scala/java/security/Timestamp.scala

```diff
@@ -1,7 +1,59 @@
 package java.security
 
-import java.util.Date
+// Ported from Harmony
+
 import java.security.cert.CertPath
+import java.util.Date
+
+import scala.collection.mutable
+
+final class Timestamp private (private val signerCertPath: CertPath,
+                               private val timestamp: Date)
+    extends Serializable {
+
+  @throws[NullPointerException]
+  def this(timestamp: Date, signerCertPath: CertPath) =
+    this(
+      {
+        if (signerCertPath == null) {
+          throw new NullPointerException("signerCertPath cannot be null")
+        }
+        signerCertPath
+      }, {
+        if (timestamp == null) {
+          throw new NullPointerException("timestamp cannot be null")
+        }
+        new Date(timestamp.getTime)
+      }
+    )
```

Comment:
I think it would be clearer if the null checks were in the primary constructor of the class. This constructor could be written as
```scala
def this(timestamp: Date, signerCertPath: CertPath) = this(signerCertPath, new Date(timestamp.getTime))
```

And the null checks would be directly in the primary constructor (that is, directly in the body of the class, below this definition for instance. Like in this example:

```scala
class A (x: String) {
  def this(x: Any) = this(x.toString)

  println("This is the primary constructor!")

  def foo = 123
  // ...
}
```

Reply:
I updated it, but we would lose the error message from before in this case...

Reply:
You can add the second null check just after the first one.

Reply:
The problem is that in the constructor there is this: `new Date(timestamp.getTime)`

Reply:
The constructor would be in this case:

```scala
  @throws[NullPointerException]
  def this(timestamp: Date, signerCertPath: CertPath) =
    this(signerCertPath, if (timestamp == null) null else new Date(timestamp.getTime()))
```

It would also have a performace penalty...

Reply:
It's perfectly acceptable to have different behavior around null-related conditions. Null pointer dereferencing is an undefined behavior in Scala Native and it doesn't need to be checked manually.

Reply:
done